vllm/entrypoints/llm.py

```diff
@@ -420,6 +425,11 @@ def encode(
             considered legacy and may be deprecated in the future. You should
             instead pass them via the ``inputs`` parameter.
         """
+        if not self.llm_engine.model_config.embedding_mode:
+            raise ValueError(
+                "LLM.encode() is only supported for embedding models."
```

Comment:
I think it is worth making these errors more specific to call out the types of HF models we need here i.e. XModel vs XForCausalLM